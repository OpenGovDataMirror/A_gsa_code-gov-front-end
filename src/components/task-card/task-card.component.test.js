import React from 'react'
import { shallow } from 'enzyme'

import { testRenderText, testRenderEmpty } from 'testUtils/render'
import TaskCard from 'components/task-card/task-card.component'

const props = {
  task: {
    title: 'task-title',
    issueURL: '/task-issue-url',
    description: 'task-desc',
    agency: {
      acronym: 'task-agency-acronym',
      name: 'task-agency-name'
    },
    date: {
      lastModified: '2020/12/10'
    }
  }
}

let wrapper
let instance
describe('components - TaskCard', () => {
  beforeEach(() => {
    wrapper = shallow(<TaskCard {...props} />)
    instance = wrapper.instance()
  })

  describe('cardTitle', () => {
    it('should render the tasks `title`', () => {
      testRenderText(instance.cardTitle, props.task.title)
    })

    it('should render a link to the tasks `issueURL`', () => {
      expect(
        shallow(instance.cardTitle)
          .find('a')
          .prop('href')
      ).toBe(props.task.issueURL)
    })

    it('should not render if the task has no `issueURL` or `title`', () => {
      wrapper.setProps({ task: { ...props.task, issueURL: undefined, title: undefined } })
      testRenderEmpty(instance.cardTitle)
    })
  })

  describe('cardDescription', () => {
    it('should render the tasks `description`', () => {
      testRenderText(instance.cardDescription, props.task.description)
    })

    it('should not render if the task has no `description`', () => {
      wrapper.setProps({ task: { ...props.task, description: undefined } })
      testRenderEmpty(instance.cardDescription)
    })
  })

  describe('agencyLink', () => {
    it('should render a custom link to the agency', () => {
      const link = shallow(<div>{instance.agencyLink}</div>).findWhere(x => x.prop('to'))
      expect(link.prop('to')).toMatch(props.task.agency.acronym)
      expect(link.prop('children')).toBe(props.task.agency.name)
    })

    it('should not render if the task has no agency `acronym` or `name`', () => {
      wrapper.setProps({ task: { ...props.task, agency: undefined } })
      testRenderEmpty(instance.agencyLink)
    })

    it('should render `Not Available` if no last modified date found', () => {
      wrapper.setProps({ task: { ...props.task, date: undefined } })
      testRenderEmpty(instance.cardDate)
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
