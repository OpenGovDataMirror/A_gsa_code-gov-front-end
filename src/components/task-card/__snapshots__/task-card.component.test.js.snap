// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - TaskCard render should render correctly 1`] = `
<li
  className="usa-card width-full margin-bottom-2"
>
  <div
    className="usa-card__container border-base-light radius-0 border-1px hover:shadow-2 height-auto card-list-item"
  >
    <header
      className="usa-card__header"
    >
      <h3
        className="usa-card__heading font-heading-lg margin-top-0"
      >
        <a
          className="issue-link"
          href="/task-issue-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          task-title
        </a>
      </h3>
    </header>
    <p
      className="width-full usa-card__body font-body-2xs"
    >
      task-desc
    </p>
    <dl
      className="width-full usa-card__body font-body-3xs padding-bottom-205 padding-top-0 border-bottom-1px border-base-light desktop:display-block display-none"
    >
      <dt
        className="display-inline text-bold margin-right-2px"
      >
        Project:
      </dt>
      <dd
        className="display-inline margin-left-2px margin-right-3"
      >
        <a
          rel="noopener noreferrer"
          target="_blank"
        />
      </dd>
      <dt
        className="display-inline text-bold margin-right-2px"
      >
        Agency:
      </dt>
      <dd
        className="display-inline margin-left-2px margin-right-3"
      >
        <Connect(CustomLinkComponent)
          to="/browse-projects?agencies=task-agency-acronym"
        >
          task-agency-name
        </Connect(CustomLinkComponent)>
      </dd>
      <dt
        className="display-inline text-bold margin-right-2px"
      >
        Last Updated:
      </dt>
      <dd
        className="display-inline margin-left-2px margin-right-3"
      >
        12/10/2020
      </dd>
    </dl>
    <div
      className="usa-card__footer font-body-3xs padding-bottom-1 padding-top-1px grid-container margin-0"
    >
      <div
        className="grid-row margin-bottom-2 tablet-lg:margin-bottom-1"
      >
        <ul
          className="display-inline-block padding-0"
        >
          <CardPart
            className="tablet:display-inline tablet:margin-right-205 display-block"
            title="Languages"
          />
          <CardPart
            className="tablet:display-inline tablet:margin-right-205 display-block"
            title="Type"
          />
          <CardPart
            className="tablet:display-inline tablet:margin-right-205 display-block"
            title="Skill Level"
          />
          <CardPart
            className="tablet:display-inline tablet:margin-right-205 display-block"
            title="Effort"
          />
        </ul>
      </div>
    </div>
  </div>
</li>
`;
